### ISSUE TITLE
Updating to latest faiss?

### ISSUE THREAD

**Issue by jondot** [Enet4/faiss-rs#72]
I've forked @SkyFan2002 's static build, and in the effort to simplify, I've also updated to latest Faiss:
https://github.com/jondot/faiss-rs
Compiling on Mac M1 and running vector search and indexing works well.

As now we have static build in *this* repo, I'm wondering if its possible to point to a recent Faiss version (and not a specific C_api flavor), or there's something hidden from my knowledge that forbids that?

**Comment by Enet4** [comment]
Updating the inner Faiss to v1.7.4 seems like a good plan for v0.13.0. If you would like to send in a PR, the steps would be:

1. Update the scripts in `faiss-sys` to point to v1.7.4
2. Run gen_bindings.sh (may need tweaking depending on the version of `bindgen` used)
3. Include changes to bindings.rs and bindings_gpu.rs
4. Adjust code at `faiss` if necessary

**Comment by jondot** [comment]
on the fork up top i just updated the git submodule without running gen_bindings or any source updates.
having said that, I selfishly built for my Mac M1 only, you're saying that if I attempt building for other platforms I might need these kind of tweaks?

**Comment by Enet4** [comment]
I'm not sure if I understood you correctly. Do you want to use a custom version of the Faiss library? That makes things a bit trickier. You will either need to depend on dynamic linking (assuming the C API is left working and compatible), or bring those changes upstream to Faiss so that they are released officially in a future version.

The changes I mentioned above are necessary to target a new version of Faiss, regardless of platform. Since Faiss does not follow semver, the crate is intended to always target a specific version of the library, or a version with a compatible C API.

**Comment by jondot** [comment]
What I'm saying is that I went ahead and updated my personal fork to this https://github.com/facebookresearch/faiss/commits/v1.7.4

And without running any binding gen steps, building statically, everything worked.


So i'm wondering -- did everything work _only_ because I built for Mac? or if I tried building for other platforms, then I would need to do `gen_bindings` etc. 

Hope that's clearer?

**Comment by Enet4** [comment]
I mean, it might have worked because of the range of API compatibility between those versions, but that may have been purely coincidental. In addition, new parts of the API are not exposed by `faiss-sys` unless the bindings are regenerated. So a complete Faiss upgrade should still go through the full process.

**Comment by web3creator** [comment]
> I've forked @SkyFan2002 's static build, and in the effort to simplify, I've also updated to latest Faiss: https://github.com/jondot/faiss-rs Compiling on Mac M1 and running vector search and indexing works well.
> 
> As now we have static build in _this_ repo, I'm wondering if its possible to point to a recent Faiss version (and not a specific C_api flavor), or there's something hidden from my knowledge that forbids that?

@jondot hello，i hava a bug on mac m1
```
  = note: ld: library not found for -lgomp
``` 
do you know what's going on'

**Comment by jondot** [comment]
yes, on mac, we need `omp` and not `gomp`, and after `omp` is specified in `build.rs`, then you need to have `libomp` on your computer.

**Comment by Enet4** [comment]
@aalekhpatel07 Thank you for reaching out. Yes, there is interest in keeping faiss-rs updated. I just need to sort out which pending pull requests should be included before the next release. Would you be willing to recreate the necessary PRs onto this repository?

**Comment by Enet4** [comment]
Your assistance is much appreciated. I have just now merged #91, so any new PRs should be based on top of this. I am unsure about the other PRs because they are rather platform specific and makes me wonder whether we can offer more control over the building process through Cargo features or environment variables.

**Comment by aalekhpatel07** [comment]
Awesome! Thanks for getting started with this. I rebased my changes onto the current master and #99 seems ready for some feedback.
